mllib/src/main/scala/org/apache/spark/ml/clustering/LDA.scala

```diff
@@ -0,0 +1,731 @@
+   *       the same.
+   *     - Values should be > 1.0
+   *     - default = uniformly (50 / k) + 1, where 50/k is common in LDA libraries and +1 follows
+   *       from Asuncion et al. (2009), who recommend a +1 adjustment for EM.
+   *  - Online
+   *     - Values should be >= 0
+   *     - default = uniformly (1.0 / k), following the implementation from
+   *       [[https://github.com/Blei-Lab/onlineldavb]].
+   * @group param
+   */
+  @Since("1.6.0")
+  final val docConcentration = new DoubleArrayParam(this, "docConcentration",
+    "Concentration parameter (commonly named \"alpha\") for the prior placed on documents'" +
+      " distributions over topics (\"theta\").", validDocConcentration)
+
+  /** Check that the docConcentration is valid, independently of other Params */
+  private def validDocConcentration(alpha: Array[Double]): Boolean = {
+    if (alpha.length == 1) {
+      alpha(0) == -1 || alpha(0) >= 1.0
+    } else if (alpha.length > 1) {
+      alpha.forall(_ >= 1.0)
+    } else {
+      false
+    }
+  }
+
+  /** @group getParam */
+  @Since("1.6.0")
+  def getDocConcentration: Array[Double] = $(docConcentration)
+
+  /**
+   * Concentration parameter (commonly named "beta" or "eta") for the prior placed on topics'
+   * distributions over terms.
+   *
+   * This is the parameter to a symmetric Dirichlet distribution.
+   *
+   * Note: The topics' distributions over terms are called "beta" in the original LDA paper
+   * by Blei et al., but are called "phi" in many later papers such as Asuncion et al., 2009.
+   *
+   * If set to -1, then topicConcentration is set automatically.
+   *  (default = -1 = automatic)
+   *
+   * Optimizer-specific parameter settings:
+   *  - EM
+   *     - Value should be > 1.0
+   *     - default = 0.1 + 1, where 0.1 gives a small amount of smoothing and +1 follows
+   *       Asuncion et al. (2009), who recommend a +1 adjustment for EM.
+   *  - Online
+   *     - Value should be >= 0
+   *     - default = (1.0 / k), following the implementation from
+   *       [[https://github.com/Blei-Lab/onlineldavb]].
+   * @group param
+   */
+  @Since("1.6.0")
+  final val topicConcentration = new DoubleParam(this, "topicConcentration",
+    "Concentration parameter (commonly named \"beta\" or \"eta\") for the prior placed on topic'" +
+      " distributions over terms.", (beta: Double) => beta == -1 || beta >= 0.0)
+
+  /** @group getParam */
+  @Since("1.6.0")
+  def getTopicConcentration: Double = $(topicConcentration)
+
+  /**
+   * Optimizer or inference algorithm used to estimate the LDA model, specified as a
+   * [[LDAOptimizer]] type.
+   * Currently supported:
+   *  - Online Variational Bayes: [[OnlineLDAOptimizer]] (default)
+   *  - Expectation-Maximization (EM): [[EMLDAOptimizer]]
+   * @group param
+   */
+  @Since("1.6.0")
+  final val optimizer = new Param[LDAOptimizer](this, "optimizer", "Optimizer or inference" +
```

Comment:
This was discussed in the design doc.  But I'm OK with reverting and flattening it out instead.  Flattening is more awkward for users IMO (since only experts need to change optimizer-specific Params), but it's easier for us developers.